src/depackers/mmcmp.c

```diff
@@ -232,8 +232,13 @@ static int block_unpack_8bit(struct block *block, struct sub_block *sub,
 	uint32 numbits = block->num_bits;
 	uint32 j, oldval = 0;
 	uint8 ptable[0x100];
+	long seekpos = hio_tell(in) + block->tt_entries;
 
-	if (hio_read(ptable, 1, 0x100, in) != 0x100) {
+	/* The way the original libmodplug depacker is written allows values
+	 * to be read from the compressed data. It's impossible to tell if this
+	 * was intentional or yet another bug. Nothing seems to rely on it. */
+	memset(ptable, 0, sizeof(ptable));
```

Comment:
do we need the memset ?

Reply:
Yes. The input stream can index any position in that array, though valid data *seems* to never index beyond `block->tt_entries`. The read is not guaranteed to initialize the entire array (which is why EVERSMOK.S3M failed to load).

Reply:
>  (which is why EVERSMOK.S3M failed to load)

The thing is, I can load and play EVERSMOK.S3M with that memset commented out

Reply:
That's because before the patch it would reject the module unless the read returned 0x100.

Reply:
OK then.